File: mifosng-android/src/main/java/com/mifos/mifosxdroid/formwidgets/FormNumericEditText.java
/*
 * This project is licensed under the open source MPL V2.
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */

package com.mifos.mifosxdroid.formwidgets;

import android.content.Context;
import android.support.annotation.IntDef;
import android.text.InputType;
import android.view.inputmethod.EditorInfo;
import android.widget.EditText;
import android.widget.TextView;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import static com.mifos.mifosxdroid.formwidgets.FormNumericEditText.FormInputTypes.TYPE_DECIMAL;
import static com.mifos.mifosxdroid.formwidgets.FormNumericEditText.FormInputTypes.TYPE_INTEGER;

/**
 * Created by ishankhanna on 01/08/14.
 */
public class FormNumericEditText extends FormWidget {

    protected TextView label;
    protected EditText input;
    protected int priority;

    @Retention(RetentionPolicy.SOURCE)
    @IntDef({TYPE_INTEGER, TYPE_DECIMAL})
    public @interface FormInputTypes {
    int TYPE_INTEGER = 0;
    int TYPE_DECIMAL = 1;
    }

    public FormNumericEditText(Context context, String property, @FormInputTypes int inputType) {
        super(context, property);

        int formInputType = inputType == FormInputTypes.TYPE_INTEGER ?
                InputType.TYPE_CLASS_NUMBER : InputType.TYPE_CLASS_PHONE;

        label = new TextView(context);
        label.setText(getDisplayText());

        input = new EditText(context);
        input.setInputType(formInputType);
        input.setImeOptions(EditorInfo.IME_ACTION_DONE);
        input.setLayoutParams(FormWidget.defaultLayoutParams);

        layout.addView(label);
        layout.addView(input);
    }

    public String getValue() {
        return input.getText().toString();
    }

    public void setValue(String value) {
        input.setText(value);
    }

    @Override
    public void setHint(String value) {
        input.setHint(value);
    }
}

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/formwidgets/FormNumericEditText.kt
/*
 * This project is licensed under the open source MPL V2.
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */
package com.mifos.mifosxdroid.formwidgets

import android.content.Context
import android.text.InputType
import android.view.inputmethod.EditorInfo
import android.widget.EditText
import android.widget.TextView

/**
 * Created by ishankhanna on 01/08/14.
 */
class FormNumericEditText(context: Context?, property: String?) : FormWidget(context, property) {
    private var label: TextView
    private var input: EditText
//    private var priority = 0

    init {
        label = TextView(context)
        label.text = displayText
        input = EditText(context)
        input.inputType = InputType.TYPE_CLASS_PHONE
        input.imeOptions = EditorInfo.IME_ACTION_NEXT
        input.layoutParams = defaultLayoutParams
        layout.addView(label)
        layout.addView(input)
    }

    override var value: String
        get() = input.text.toString()
        set(value) {
            input.setText(value)
        }

    override fun setHint(value: String?) {
        input.hint = value
    }
}